whatweb

```diff
@@ -366,28 +329,27 @@ def open_target( url )
 end
 
 ### output
-
-def error( s )
-	return if $NO_ERRORS
-	if defined?( $semaphore )
-		# We want the output mutex locked.
-		# Has our current thread already locked the Mutex?
-		begin
-			$semaphore.lock
-		rescue ThreadError
-			# we're already locked. This was expected.
```

Comment:
If this happens, the code continues on to unlock when it shouldn't.

Reply:
Thanks for spotting this error @Code0x58